Week1/homework/app.js

```diff
@@ -1,11 +1,123 @@
 'use strict';
 {
-  const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets'
+  ///2.1
+  const bookTitles  = [
+    'harry_potter_chamber_secrets',
+    'the_Hunger_Games',
+    'to_kill_a_mockingbird',
+    'Twilight',
+    'The_Book_Thief',
+    'Crime_and_Punishment',
+    'The_Princess_Bride',
+    'game_of_thrones',
+    'Brave_New_World',
+    'The_Outsiders'
   ];
 
+  ///2.3
+  function genBookList(ul) {
+    for(let i = 0 ; i < bookTitles .length; i++){
+    const li = document.createElement('li');
+    ul.appendChild(li);
+    li.innerText = bookTitles[i] ;
+    }
+  }
+
+  function main() {
+    const root = document.getElementById('root');
+    const h1 = document.createElement('h1');
+    h1.innerText = 'My books';
+    root.appendChild(h1);
+    const ul = document.createElement('ul');
+    root.appendChild(ul);
+    genBookList(ul);
+  }
+
+  window.addEventListener('load', main);
+
+
+  ///2.4
+  const bookInfo = {
+    'harry_potter_chamber_secrets':{title:'harry potter chamber secrets',language:'english',author:'zzz'},
+    'the_Hunger_Games':{title:'the Hunger Games',language:'english',author:'zzz'},
+    'to_kill_a_mockingbird':{title:'to kill a mockingbird',language:'english',author:'zzz'},
+    'Twilight':{title:'Twilight',language:'english',author:'zzz'},
+    'The_Book_Thief':{title:'The Book Thief',language:'english',author:'zzz'},
+    'Crime_and_Punishment':{title:'Crime and Punishment',language:'english',author:'zzz'},
+    'The_Princess_Bride':{title:'The Princess Bride',language:'english',author:'zzz'},
+    'game_of_thrones':{title:'game of thrones',language:'english',author:'zzz'},
+    'Brave_New_World':{title:'Brave New World',language:'english',author:'zzz'},
+    'The_Outsiders':{title:'The Outsiders',language:'english',author:'zzz'}
+  };
+
+  ///2.5
+  function genBookInfo(){
+    const root = document.getElementById('root');
+    const h1 = document.createElement('h1');
+    h1.innerText = 'My books details';
+    root.appendChild(h1);
+    const ul = document.createElement('ul');
+    root.appendChild(ul);
+    const opjectsIndex = Object.keys(bookInfo);
+    for(let i = 0; i < opjectsIndex.length; i++){
```

Comment:
In this `for` loop you can iterate through the `bookTitles` array and add book info details to the existing `li`-s that you create in the first loop. You can get a reference to each `li` by using `getElementById()`, assuming you gave each `li` an `id` in the previous loop.